llama_index/indices/query/query_transform/feedback_transform.py

```diff
@@ -1,37 +1,105 @@
+import logging
 from typing import Dict, Optional
 
 from llama_index.evaluation.base import Evaluation
 from llama_index.indices.query.query_transform.base import BaseQueryTransform
 from llama_index.indices.query.schema import QueryBundle
+from llama_index.langchain_helpers.chain_wrapper import LLMPredictor
+from llama_index.llm_predictor.base import BaseLLMPredictor
+from llama_index.prompts.base import Prompt
+from llama_index.response.schema import Response
+
+logger = logging.getLogger(__name__)
+
+DEFAULT_RESYNTHESIS_PROMPT_TMPL = (
+    "Here is the original query:\n"
+    "{query_str}\n"
+    "Here is the response given:\n"
+    "{response}\n"
+    "Here is some feedback from evaluator about the response given.\n"
+    "{feedback}\n"
+    "If you want to resynthesize the query, please return the modified query below.\n"
+    "Otherwise, please return the original query.\n"
+)
+
+DEFAULT_RESYNTHESIS_PROMPT = Prompt(DEFAULT_RESYNTHESIS_PROMPT_TMPL)
 
 
 class FeedbackQueryTransformation(BaseQueryTransform):
     """Transform the query given the evaluation feedback.
 
     Args:
         eval(Evaluation): An evaluation object.
+        llm_predictor(BaseLLMPredictor): An LLM predictor.
+        resynthesize_query(bool): Whether to resynthesize the query.
+        resynthesis_prompt(Prompt): A prompt for resynthesizing the query.
 
     """
 
     def __init__(
         self,
-        evaluation: Evaluation,
+        evaluation: Optional[Evaluation] = None,
```

Comment:
nit: let's pass it at run time, rather than keep as a state.

Reply:
Changed